### ISSUE TITLE
Axios fails to follow redirects from MSW mocked responses

### ISSUE THREAD

**Issue by a11rew** [kettanaito/msw#2177]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

v20.12.2

### Other versions
`msw` - 2.3.1
`axios` - 1.7.2

### Reproduction repository

https://codesandbox.io/p/devbox/axios-follow-redirects-issue-9qn7z9

### Reproduction steps

1. Setup a http.get response handler that responds with a redirect (HttpResponse.redirect or `new Response` with 3xx code and Location header)
2. Make a GET request to the URL specified in the handler

### Current behavior

The request fails with this error message originating from the `follow-redirects` library Axios uses
```
Error [ERR_FR_REDIRECTION_FAILURE]: Redirected request failed: Cannot read properties of undefined (reading 'getHeader')
```

The stack trace points to this line in the `follow-redirects` lib
https://github.com/follow-redirects/follow-redirects/blob/35a517c5861d79dc8bff7db8626013d20b711b06/index.js#L422

`response.req.getHeader` is undefined in the `IncomingMessage`/response follow-redirects receives when the response is mocked using MSW. This does not happen when the requests returning redirects are not intercepted with MSW.

### Expected behavior

Axios should be able to follow redirects normally like it does when responses are not mocked with MSW.

**Comment by kettanaito** [comment]
Hi, @a11rew. Thanks for reporting this. 

We have a test for `follow-redirects` [here](https://github.com/mswjs/interceptors/blob/e822e14bc669a7e5cefa9b9b0f627ea37c76ba99/test/third-party/follow-redirect-http.test.ts#L43) but, apparently, it misses something. One difference I can spot is that in our tests the original server responds with a redirect, while in your use case the mock does that. Perhaps that's the culprit. 

Would you have some time to add a new test case to the test I've linked above? Something like

```js
interceptor.on('request', ({ request }) => {
  request.respondWith(new Response(null, {
    status: 300,
    headers: { Location: '/new/url' }
  }))
})
```

**Comment by a11rew** [comment]
Added the failing test case in this PR https://github.com/mswjs/interceptors/pull/581  @kettanaito. I'd be open to contributing a fix for this, any pointers on where I should be looking?

**Comment by kettanaito** [comment]
Could be potentially fixed by https://github.com/mswjs/interceptors/issues/626. Afaik, Axios uses XHR by default, so we need to look into whether XHR follows redirects to distinguish if that's a request client behavior or Axios' behavior.

**Comment by kettanaito** [comment]
## Released: v2.4.4 🎉

This has been released in v2.4.4!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.4.4)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.4.4)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
